hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/FileSystem.java

```diff
@@ -4876,6 +4876,16 @@ public CompletableFuture<FSDataInputStream> build() throws IOException {
 
   }
 
+  /**
+   * Return root path
+   * @param path
+   * @return
+   * @throws IOException
+   */
+  public Path getEnclosingRoot(Path path) throws IOException {
+    return this.makeQualified(new Path("/"));
```

Comment:
MUST verify the path passed in is part of the current FS, so call makeQualified on it too, even if result is discarded. plus javadoc to explain

Reply:
done